src/components/data_search/DatasetSearchTable.js

```diff
@@ -28,8 +29,42 @@ const datasetTableHeader = [
 ];
 
 export const DatasetSearchTable = (props) => {
-  const { datasets, isLoading } = props;
+  const { datasets, isLoading, history } = props;
   const [tableData, setTableData] = useState({});
+  const [selected, setSelected] = useState([]);
+
+  const selectHandler = (event, data, selector) => {
+    let idsToModify = [];
+    if (selector === 'all') {
+      data.rows.forEach((row) => {
+        row.subtable.rows.forEach((subRow) => {
+          idsToModify.push(subRow.id);
+        });
+      });
+    } else if (selector === 'row') {
+      data.subtable.rows.forEach((row) => {
+        idsToModify.push(row.id);
+      });
+    } else if (selector === 'subrow') {
+      idsToModify.push(data.id);
+    }
+
+    let newSelected = [];
+    const allSelected = idsToModify.every((id) => selected.includes(id));
+    if (allSelected) {
+      newSelected = selected.filter((id) => !idsToModify.includes(id));
+    } else {
+      newSelected = selected.concat(idsToModify);
+    }
+
+    setSelected(newSelected);
+  };
+
+  const applyForAccess = async () => {
+    const draftDatasets = selected.map((id) => parseInt(id.replace('dataset-', '')));
+    const darDraft = await DAR.postDarDraft({ datasetId: draftDatasets });
```

Comment:
Not your code, but I'm confused about why the singular `datasetId` is used for a list of ids. Probably a legacy code issue. Maybe we can clean this up in a different ticket.

Reply:
I created a ticket for this: https://broadworkbench.atlassian.net/browse/DUOS-2652